### DISCUSSION TITLE
Support for TRS-80 DMK image files

### DISCUSSION THREAD

**Discussion by malmacleod** [keirf/greaseweazle#393] [Ideas]
Hello Keir, 

I am a relatively new user of the Greaseweazle, but have been very impressed with it so far.  I've also been using ImageDisk and Catweasel for many years.

My main use at the moment for the Greaseweazle is to archive and recreate 5.25" FM/MFM disks for the TRS-80 Models 1, 3 & 4.  I've been doing this successfully with DSK and IMD files, but it would be great if support for DMK could also be added.  DSK in particular is not well suited for the TRS-80, because of the non-standard DAMs used in the directory track (I'm aware that you dealt with this issue last year for the TRS-80 Model 1) and of course this metadata is not stored in DSK files (which in the TRS80 world are sometimes called JV1 files).

DMK was used by some early PC-based TRS-80 emulators such as TRS32 (and DMK is still supported by current emulators such as TRS80gp), and there are many online repositories for TRS-80 disks that are only in DMK format (often these are named as DSK files, but on examination of the first 4 bytes they can be verified as being DMK files).

Would it be possible to add support for DMK please?  If so, that would lessen the dependence that I currently have to use Catweasel for this task.  FYI the DMK spec is available here -> http://cpmarchives.classiccmp.org//trs80/mirrors/www.discover-net.net/~dmkeil/trs80/trstech.htm#Technical-DMK-disks

Secondary question: Are there any other file formats (ie not DSK or IMD) already supported by Greaseweazle that are well suited for FM/MFM-formatted disks?  By that I mean containers that have good metadata storage for DAMs, interleave, etc.

An example of a DMK file for the TRS-80 is LS-DOS 6.3.1H on this page -> https://www.tim-mann.org/misosys.html

If you need any further info or examples, I'm happy to help!

Mal

**Comment by qbarnes** [comment]
This might be considered a duplicate of #306 .

There's two independent issues here.  One is just the support of the DMK file format.  The other is supporting a variety of the floppy disk formats that were generated by the 8-bit operating systems of the day.  The latter involves unusual DAM use, mixed formats (FM and MFM between different tracks and sometimes even within the same track), etc., etc.  Back in the day, floppy disk formats were much more of a wild west.
